### ISSUE TITLE
Hatchling fails to build annotated-types with "Field `project.license-files` must be an array"

### ISSUE THREAD

**Issue by shoeffner** [annotated-types/annotated-types#85]
Hatchling [introduced a check](https://github.com/pypa/hatch/commit/28f233c535508247ffa9a40dab82c1d1cb2b700b#diff-b29708f9e31a5480d8552a96951ee054262d2d12b906e3997413eacc99f33b90R751-R753) for the project.license-files key to be [PEP-639](https://peps.python.org/pep-0639/) compliant. This means that license-files can no longer be
```
license-files = { paths = ['LICENSE'] }
```
but instead should be `license-files = ["LICENSE"]` or simply left out as [the default will glob](https://github.com/pypa/hatch/blob/c1a3c6b4c86ea2affe4734b54578f032fe5d0003/backend/src/hatchling/metadata/core.py#L748-L758) for that file:
```python
globs = ['LICEN[CS]E*', 'COPYING*', 'NOTICE*', 'AUTHORS*']
```

If you like, I can provide a PR with either of the two solutions.

**Comment by Zac-HD** [comment]
Thanks for reporting this!  I'd be happy to switch to `license-files = ["LICENSE"]` 🙂

**Comment by shoeffner** [comment]
Turns out that hatch 1.26.1 released a compatibility fix on the same day our builds failed, which is the reason I was not able to replicate the behavior locally.

https://github.com/pypa/hatch/compare/hatchling-v1.26.0...hatchling-v1.26.1#diff-b29708f9e31a5480d8552a96951ee054262d2d12b906e3997413eacc99f33b90R746-R747

Since the non-compatible version has been yanked, I still was not able to reproduce this properly as I don't know the exact versions used.
Lessons learned: We will now run a pip freeze after (successful and unsuccessful) builds so that we at least know which tools we actually used.
I'll still create the PR :)
